Mathlib/Data/List/Basic.lean

```diff
@@ -771,6 +771,9 @@ theorem getLast?_append {l₁ l₂ : List α} {x : α} (h : x ∈ l₂.getLast?)
 
 /-! ### head(!?) and tail -/
 
+@[simp]
+theorem head!_nil : List.head! [] = 0 := rfl
```

Comment:
Do we have a good way of making this more polymorphic? I guess we don't have a type class for `default = 0`, but does simp already turn the default Nat into 0? If so we could generalise this

Reply:
We have `@[simp] Int.default_eq_zero`, but no `Nat.default_eq_zero`. I agree we should generalise
```suggestion
theorem head!_nil : ([] : List α).head! = default := rfl
```